### PR TITLE
add __cudaInitModule to cudart_stub.cc

### LINKED ISSUE #39280: undefined reference to `__cudaInitModule' when compiling run_hlo_module in debug mode
Build is configured with CUDA.

When compiling run_hlo_module as follows:
 
`
bazel build --config=dbg -j 24 --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0" //tensorflow/compiler/xla/tools:run_hlo_module
`

it results in the following error:

```
ERROR: /home/baarts/tensorflow/tensorflow/tensorflow/compiler/xla/tools/BUILD:96:1: Linking of rule '//tensorflow/compiler/xla/tools:replay_computation_gpu' failed (Exit 1)
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f64_reduce_scatter.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f64_reduce.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f64_all_reduce.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f32_reduce_scatter.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f32_reduce.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f32_all_reduce.cu.pic.o):/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: more undefined references to `__cudaInitModule' follow
collect2: error: ld returned 1 exit status
Target //tensorflow/compiler/xla/tools:replay_computation_gpu failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 65.233s, Critical Path: 55.19s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

the rule that generates libdevice_a.a includes host_runtime.h which contains the following code:

```
static char __nv_init_managed_rt_with_module(void **handle)
{
  return __cudaInitModule(handle);
}
```

Because this is  a debug compilation, this function is not removed, and pulls in a reference to __cudaInitModule. This function is missing from tensorflow/stream_executor/cuda/cudart_stub.cc

The same error is there when compiling replay_computation_gpu and interactive_graphviz

### PR DESCRIPTION
**Pull request by bas-aarts**
this fixes https://github.com/tensorflow/tensorflow/issues/39280

### CODE REVIEW

File: tensorflow/stream_executor/cuda/cudart_stub.cc
Comment (yifeif) [inline tensorflow/stream_executor/cuda/cudart_stub.cc:135]: return type should be char

File: tensorflow/stream_executor/cuda/cudart_stub.cc
Comment (yifeif) [inline tensorflow/stream_executor/cuda/cudart_stub.cc:137]: same here, but I'm not sure what char to return if not found. @chsigg

Comment (chsigg) [reply tensorflow/stream_executor/cuda/cudart_stub.cc:137]: 0, 1, or π
;-)

It's not documented, but probably Bastiaan has access to the implementation.

File: tensorflow/stream_executor/cuda/cudart_stub.cc
Comment (bas-aarts) [reply tensorflow/stream_executor/cuda/cudart_stub.cc]: done. too eager with copy-paste

File: tensorflow/stream_executor/cuda/cudart_stub.cc
Comment (bas-aarts) [reply tensorflow/stream_executor/cuda/cudart_stub.cc]: done. too eager with copy-paste.
0 on failure

File: tensorflow/stream_executor/cuda/cudart_stub.cc
Comment (yifeif) [reply tensorflow/stream_executor/cuda/cudart_stub.cc]: I thought that might have been the case :). LG!

File: tensorflow/stream_executor/cuda/cudart_stub.cc
Comment (yifeif) [reply tensorflow/stream_executor/cuda/cudart_stub.cc]: Thanks @bas-aarts and @chsigg!

**Review by yifeif — APPROVED** [review]

**Comment by bas-aarts** [comment]
Anything I can do on my end to get this merged?

**Comment by gbaned** [comment]
@bas-aarts Nothing to do at this moment. It is waiting for internal approval. We will come back if anything required. Thanks!

**Pull request merged**
